### PR TITLE
[MHV-45980] Added Remove-Attachment Modal, 59491

### PR DESCRIPTION
**Pull request by vsaleem**
## Summary

- Added Remove Attachments modal
- This implementation covers improving the [User flows and navigation], and the accessibility issue that - MUST - User should be able to review, edit, or confirm transactional information before submission.
- The Platform recommendation used for this ac was option 2. Add an intermediate confirmation step between a user clicking 'Delete/Remove' and the issue actually being removed.
- Before, a user using a screenreader did not receive any notice that their attachments were deleted after selecting the remove button.
- Now, the screen-reader will report a status that their attachment file has been removed after following the modal prompts.

## Related issue(s)

JIRA:
- jira.devops.va.gov/browse/MHV-45980

Github Issue: 
- github.com/department-of-veterans-affairs/va.gov-team/issues/59491

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |image 
image|

## What areas of the site does it impact?

*Va.gov - Secure Messaging*

## Acceptance criteria

1) When an attachment is attached to the UI, then the focus should read the file name on the card.
2) The "Remove" button on attachment should have an aria-label that reads, "Remove [file name]".
3) After the Remove button is selected, a modal appears: user confirms remove modal.
4) After user confirms removing the attachment, the attachment card disappears and focus lands on the "Attach File" button, and the screenreader status is announced that "File [file name] successfully removed."

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by mdewey — APPROVED** [review]
LGTM

File: src/applications/mhv/secure-messaging/components/AttachmentsList.jsx [hunk 1]
```diff
@@ -64,16 +68,29 @@ const AttachmentsList = props => {
...
+                    />
                     <span>{file.name} </span>(
                     {getSize(file.size || file.attachmentSize)})
                   </span>
-                  <va-button
-                    onClick={() => removeAttachment(file)}
-                    secondary
-                    text="REMOVE"
+                  <button
+                    type="button"
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/AttachmentsList.jsx:80]: is `type="button"` needed on `<button>` element?

Comment (vsaleem) [reply src/applications/mhv/secure-messaging/components/AttachmentsList.jsx:91]: I think we still need this.. it is an explicit type and might upset the liner if we don't have it... :eyes:

File: src/applications/mhv/secure-messaging/components/ComposeForm/DraftSavedInfo.jsx [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
 
   const { isSaving, lastSaveTime, saveError } = useSelector(
     state => state.sm.draftDetails,
   );
+  const [ariaLive, setAriaLive] = useState('off');
+
+  useEffect(
+    () => {
+      setAriaLive(attachments.length > 0 ? 'off' : 'polite');
+    },
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/ComposeForm/DraftSavedInfo.jsx:18]: I think `useMemo` hook would be more appropriate here

**Pull request closed without merging**